pkg/bpf/map_linux_test.go

```diff
@@ -87,16 +113,270 @@ func (s *BPFPrivilegedTestSuite) TestOpen(c *C) {
 	// existingMap is the same as testMap. Opening should succeed.
 	existingMap := NewMap("cilium_test",
 		MapTypeHash,
-		4,
-		4,
+		int(unsafe.Sizeof(TestKey{})),
+		int(unsafe.Sizeof(TestValue{})),
 		maxEntries,
 		BPF_F_NO_PREALLOC,
 		0,
-		nil)
+		DumpParserFunc).WithCache()
 	err = existingMap.Open()
 	c.Check(err, IsNil)      // Avoid assert to ensure Close() is called below.
 	err = existingMap.Open() // Reopen should be no-op.
 	c.Check(err, IsNil)
 	err = existingMap.Close()
 	c.Assert(err, IsNil)
 }
+
+func (s *BPFPrivilegedTestSuite) TestOpenMap(c *C) {
+	openedMap, err := OpenMap("cilium_test_no_exist")
+	c.Check(err, Not(IsNil))
+	c.Check(openedMap, IsNil)
+
+	openedMap, err = OpenMap("cilium_test")
+	c.Check(err, IsNil)
+}
+
+func (s *BPFPrivilegedTestSuite) TestOpenOrCreate(c *C) {
+	// existingMap is the same as testMap. OpenOrCreate should skip recreation.
+	existingMap := NewMap("cilium_test",
+		MapTypeHash,
+		int(unsafe.Sizeof(TestKey{})),
+		int(unsafe.Sizeof(TestValue{})),
+		maxEntries,
+		BPF_F_NO_PREALLOC,
+		0,
+		DumpParserFunc).WithCache()
+	isNew, err := existingMap.OpenOrCreate()
+	c.Check(err, IsNil) // Avoid assert to ensure Close() is called below.
```

Comment:
For future reference, we often defer the `Close()` to handle cases like this. This is fine as-is, just letting you know in case you want to use that approach in future.

Reply:
IIUC all the `Check` I introduced could be changed to `Assert` with it, is that right? I didn't have much consideration for this so will fix all of these.

Reply:
I think so, yeah. So long as cleanup is deferred it should be ok to use assert.

Reply:
In my last force-push I replaced all with `Assert` + `defer Close()` where testing `Close()` is unnecessary.